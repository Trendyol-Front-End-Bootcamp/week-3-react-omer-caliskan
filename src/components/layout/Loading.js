import React from'react';
import Pickle from "../../assets/images/pickle.png";
import '../../assets/css/loading.css';

function Loading(){
    return (
        <div className="loading">
            <img src={Pickle} alt="Wubba Lubba Dub Dub"/>
            <h2>Page Loading ...</h2>
        </div>
    )
}

export default Loading;