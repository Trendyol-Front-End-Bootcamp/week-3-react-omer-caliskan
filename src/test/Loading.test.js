import Pickle from "../../assets/images/pickle.png";
import React from 'react'
import Loading from "../components/layout/Loading";
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() })
describe("Loading Component Test", () => {
        let loading
        beforeEach(() => {
          loading = shallow( <Loading />);
        })
  
        it("should render loading", () => {
          expect(loading.debug()).toMatchSnapshot();
        })
  
        it("should have true class", () => {
          const classTruthy = loading.hasClass("loading")
  
          expect(classTruthy).toBeTruthy()
        })
  
        it("should have true img", () => {
  
          const imageSrc = loading.childAt(0).prop('src') 
  
          expect(imageSrc).toEqual(Pickle)
        })
  
        it("should have true text", () => {
  
          const loadingText = loading.childAt(1).find('h2')
          
          expect(loadingText.text()).toEqual('Page Loading ...')
        })
      })