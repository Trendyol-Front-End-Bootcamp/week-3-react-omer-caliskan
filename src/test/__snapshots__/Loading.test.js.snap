// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Test should render loading 1`] = `
"<div className=\\"loading\\">
  <img src={{...}} alt=\\"Wubba Lubba Dub Dub\\" />
  <h2>
    Page Loading ...
  </h2>
</div>"
`;
